bin/auth_ejabberd.php

```diff
@@ -51,9 +51,10 @@
  *
  */
 
+use Friendica\Network\HTTPException\ForbiddenException;
+
 if (php_sapi_name() !== 'cli') {
-	header($_SERVER["SERVER_PROTOCOL"] . ' 403 Forbidden');
-	exit();
+	throw new ForbiddenException();
```

Comment:
These exceptions can't be used outside of the Friendica App context because there we catch them and pass them to `Module\Special\HTTPException::rawContent` which sets the correct HTTP response code. Without this catch it will just trigger the default webserver 500 error when PHP encounters an uncaught exception.

Reply:
How could we use some centralized function here instead of the old code? I find it rather unelegant  using a hardcoded `header`here.

Reply:
You can use `System::httpExit` for this.

Reply:
This calls the logger. Wouldn't this fail as well?

Reply:
Only one way to know: try for yourself.

Reply:
`System::httpExit` didn't work?

Reply:
I haven't tried.